databases/week1/main.sql

```diff
@@ -0,0 +1,127 @@
+----1. Find out how many tasks are in the task table
+SELECT
+    count(id)
+FROM
+    task;
+
+---2. Find out how many tasks in the task table do not have a valid due date
+SELECT
+    *
+FROM
+    task
+WHERE
+    due_date is NULL;
+
+---3. Find all the tasks that are marked as done
+SELECT
+    task.id,task.title,status.name
+FROM
+    task
+INNER JOIN
+    status
+ON
+    task.status_id=status.id 
+WHERE
+    status.name = 'Done';
+
+---4. Find all the tasks that are not marked as done
+SELECT
+    task.id,task.title,status.name
+FROM
+    task
+INNER JOIN
+    status
+ON
+    task.status_id=status.id 
+WHERE
+    status.name != 'Done';
+
+---5. Get all the tasks, sorted with the most recently created first
+SELECT
+    task.id, 
+    task.title, 
+    task.created
+FROM
+    task
+ORDER BY
+    task.created DESC;        
+
+---6. Get the single most recently created task
+--- i. Using RANK() which is most useful for getting only the task of any rank:
+SELECT
+*
+FROM
+    (SELECT
+        task.id, 
+        task.title, 
+        task.created,
+        RANK() OVER (
+            ORDER BY task.created desc
```

Comment:
Impressive (Y), but this also does the job

 SELECT 
    *
FROM
    task
ORDER BY
    created
DESC
LIMIT 1;